File: Python/Laboratorio-4/Lab04Ejercicio2.py
#
# Lab04Ejercicio2.py
# 
# Descripcion: Este programa calcula la suma de todos los elementos de 
# una matriz NxM
#
# Autor: Douglas Torres
# Ultima modificacion: 06/05/2014
#
# Variables:
#    matriz : array // Arreglo que guarda la matriz NxM
#    N : integer // Variable que dice la cantidad de filas de la matriz
#    M : integer // Variable que dice la cantidad de columnas de la matriz
#    suma : integer // Variable que guarda la suma de los elementos de la matriz

# Entrada de datos

N = int(input("Introduzca el numero de filas que desea en su matriz: "))

M = int(input("Introduzca el numero de columnas que desea en su matriz: "))

matriz = [[int(input("matriz[" + str(i) +","+str(j)+"]=")) for i in range (N)] \
			for j in range (M)]
# Precondicion

try:
	assert (N>0 and M>0 and len(matriz)>0)
except AssertionError:
	print ("La precondicion no se cumple.")
	quit()

# Calculo de datos

suma=0

for i in range (N):
	for j in range (M):
		suma = suma + matriz[i][j]

# Postcondicion

try:
	assert (suma>=0)
except AssertionError:
	print ("La postcondicion no se cumple.")
	quit()

# Salida de datos

print ("La suma de todos los elemento de la matriz es =",suma)

File: Pascal/lab-5/Lab04Ejercicio2.pas
(*
 *  Ejercicio 2 del laboratorio 4
 *
 *  Descripcion: El siguiente programa pide al usuario los coeficientes
 *  de un polinomio y el valor de la X y se imprime la evaluacion
 *  Autor: Douglas Torres
 *  Ultima modificacion: 21/03/2014
 *)

program Lab04Ejercicio2;

const
   N = 5;
   
type
   polinomio = array [0..N] of real;

var
   C	     : polinomio;
   resultado : real;
   i	     : integer;
   x	     : real;
   expp	     : real;
	     
begin

   (* Entrada de datos*)

   writeln;
   writeln ('Introduzca los coeficientes de un polinomio');
   for i := 0 to N do
   begin
      write ('Coeficiente de grado ',i:2, ' = ');
      readln (C[i])
   end;
   writeln ('De un valor de X para calcular el valor del polinomio');
   readln (x);

   {Precondicion: N,C %in Z /\ x %in R /\ (%forall i : 0<=i<=N : C[i])}

   (* Calculo de datos *)
   resultado := 0;

   for i := 0 to N do
   begin
      expp := Exp(LN(x)*i);
      resultado := resultado+(C[i]*expp);
   end;
   
   {Postcondicion: resultado in R}
   
   (* Salida de resultados *)

   writeln;
   writeln ('El resultado es');
   writeln ('P(',x:0:2,') = ',resultado:0:2);
   readln;
end.